File: backend/app/schemas/audit.py
"""Audit log read schemas."""

from __future__ import annotations

from datetime import datetime
from typing import Any, Optional

from pydantic import BaseModel, ConfigDict


class AuditEntry(BaseModel):
    """A single audit-trail entry (read model)."""

    model_config = ConfigDict(from_attributes=True)

    id: str
    action: str
    actor_id: Optional[str] = None
    organization_id: Optional[str] = None
    entity_type: Optional[str] = None
    entity_id: Optional[str] = None
    meta: Optional[dict[str, Any]] = None
    created_at: datetime

File: frontend/src/types/audit.ts
export interface AuditEntry {
  id: string;
  action: string;
  actor_id: string | null;
  organization_id: string | null;
  entity_type: string | null;
  entity_id: string | null;
  meta: Record<string, unknown> | null;
  created_at: string;
}

export interface AuditPage {
  items: AuditEntry[];
  total: number;
  limit: number;
  offset: number;
}
